src/components/RecipeModal.tsx

```diff
@@ -0,0 +1,121 @@
+import { FC, useState, ChangeEvent } from "react";
+import { useHistory } from "react-router-dom";
+import { styled } from "styled-components";
+import { useQueryClient } from "react-query";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from "@mui/material";
+
+import {
+  QUERY_KEYS,
+  Recipe,
+  URL_PATHS,
+  useCreateRecipe,
+  useRecipe,
+  useRecipes,
+  useUpdateRecipe,
+} from "../api";
+
+interface RecipeModalProps {
+  open: boolean;
+  onClose: () => void;
+  recipe?: Recipe;
+}
+export const RecipeModal: FC<RecipeModalProps> = ({
```

Comment:
There's no need anymore to type the react components with the `FC` typing, you can do directly
`export const RecipeModal = ({ open, onClose, recipe: initialRecipe } = props) => {.....}`

Reply:
Yeah thats fair, just been used to typing the generic for a while, and seems TK uses 
```
const ComponentA = (props: Props) => 
```
over
```
const ComponentB: FC<Props> = (props) =>
```
So will use ComponentA format going forward!